rust/src/ike/ikev1.rs

```diff
@@ -0,0 +1,158 @@
+    tx.hdr.ikev1_header.maj_ver = Some(isakmp_header.maj_ver);
+    tx.hdr.ikev1_header.min_ver = Some(isakmp_header.min_ver);
+    tx.hdr.ikev1_header.exchange_type = Some(isakmp_header.exch_type);
+
+    let mut cur_payload_type = isakmp_header.next_payload;
+    let mut payload_types: HashSet<u8> = HashSet::new();
+    payload_types.insert(cur_payload_type);
+
+    if isakmp_header.flags & 0x01 != 0x01 {
+        match parse_ikev1_payload_list(current) {
+            Ok((rem, payload_list)) => {
+                for isakmp_payload in payload_list {
+                    if let Err(_) = parse_payload(
+                        cur_payload_type,
+                        isakmp_payload.data,
+                        isakmp_payload.data.len() as u16,
+                        &mut state.ikev1_container.domain_of_interpretation,
+                        &mut tx.hdr.ikev1_header.key_exchange,
+                        &mut tx.hdr.ikev1_header.nonce,
+                        &mut tx.hdr.ikev1_transforms,
+                        &mut tx.hdr.ikev1_header.vendor_ids,
+                        &mut payload_types
+                    ) {
+                        SCLogDebug!("Error while parsing IKEV1 payloads");
+                        return AppLayerResult::err();
+                    }
+
+                    cur_payload_type = isakmp_payload.payload_header.next_payload;
+                }
+
+                if payload_types.contains(&(IsakmpPayloadType::SecurityAssociation as u8)) {
+                    // clear transforms on a new SA in case there is happening a new key exchange
+                    // on the same flow, elsewise properties would be added to the old/other SA
+                    if direction == STREAM_TOSERVER {
+                        state.ikev1_container.client.reset();
+                    } else {
+                        state.ikev1_container.server.reset();
+                    }
+                }
+
+                // add transaction values to state values
+                if direction == STREAM_TOSERVER {
+                    state.ikev1_container.client.update(
+                        &tx.hdr.ikev1_header.key_exchange,
+                        &tx.hdr.ikev1_header.nonce,
+                        &tx.hdr.ikev1_header.vendor_ids,
+                        &tx.hdr.ikev1_transforms
+                    );
+                } else {
+                    state.ikev1_container.server.update(
+                        &tx.hdr.ikev1_header.key_exchange,
+                        &tx.hdr.ikev1_header.nonce,
+                        &tx.hdr.ikev1_header.vendor_ids,
+                        &tx.hdr.ikev1_transforms
+                    );
+                }
+
+                if rem.len() > 0 {
+                    // more data left unread than should be
+                    SCLogDebug!("Unread Payload Data");
+                    state.set_event(IkeEvent::PayloadExtraData);
```

Comment:
There should be a files in the `rules` directory to match on these app-layer events